### DISCUSSION TITLE
Tagging empty constraints

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#1066] [General discussion] [answered]
Consider the following type of constraint:

`( x( bActive ) <= 3 ) : 'x<3'`

Sometimes, `x( bActive )` is empty, which triggers an assert:

`For colon operator with char operands, first and last operands must be char.`

Probably because it's empty, it doesn't overload the sdp colon operator. This means that I need to check first `isempty( x( bActive ) )`. If I have a set of constraints like that, then I'll need to check each one separately, and I couldn't write them together.

Do you perhaps have a solution for this?

Coming to think of it, I can define a function `cons( con, label )`, which would check if con is empty and return `[]` if it is.

On the subject, sometimes I need to surround the constraints in parentheses (like I did in the example above), or they are interpreted wrong. So, I always put them in parentheses.

**Comment by johanlofberg** [comment, marked as answer]
As you say, impossible to get around as the use of : here is a hackish overloading which doesn't apply to anything but constraint objects
